### PR TITLE
added tensor serialization

### PR DESCRIPTION
**Pull request by vkkhare**
WIP tensor serialization.

### CODE REVIEW

**Comment by vkkhare** [comment]
I had to also solve #57 as I anyway had to deal with nullability of models everywhere. Going by your suggestion @mccorby and removed `lateinit` 
The PR should close #30

File: demo-app/src/main/java/org/openmined/syft/demo/domain/MNISTDataRepository.kt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package org.openmined.syft.demo.domain
 
 import org.openmined.syft.demo.datasource.LocalMNISTDataDataSource
-import java.util.ArrayList
+import java.util.*
```
Comment (mccorby) [inline demo-app/src/main/java/org/openmined/syft/demo/domain/MNISTDataRepository.kt]: Keep the specific import.
I think this has to be set in the preferences.

File: syftlib/src/main/java/org/openmined/syft/execution/JobStatusSubscriber.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    fun onJobStatusMessage(jobStatusMessage: JobStatusMessage) {
+        when (jobStatusMessage) {
+            is JobStatusMessage.JobReady -> {
+                if (jobStatusMessage.clientConfig != null)
+                    onReady(
+                        jobStatusMessage.model,
+                        jobStatusMessage.clientConfig
+                    )
+                else
+                    onError(IllegalStateException("Client config not avialable yet"))
```
Comment (mccorby) [inline syftlib/src/main/java/org/openmined/syft/execution/JobStatusSubscriber.kt]: Typo: available

File: syftlib/src/main/java/org/openmined/syft/proto/SyftTensor.kt [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+import org.openmined.syftproto.types.syft.v1.IdOuterClass
+import org.openmined.syftproto.types.torch.v1.SizeOuterClass
+import org.openmined.syftproto.types.torch.v1.Tensor
+import org.openmined.syftproto.types.torch.v1.TensorDataOuterClass
+import org.pytorch.DType
+import java.io.InvalidClassException
+import java.lang.Math.random
+import java.nio.DoubleBuffer
+import java.util.*
```
Comment (mccorby) [inline syftlib/src/main/java/org/openmined/syft/proto/SyftTensor.kt]: Same as above

File: demo-app/src/main/java/org/openmined/syft/demo/domain/MNISTDataRepository.kt (hunk 1 above)
Comment (vkkhare) [reply demo-app/src/main/java/org/openmined/syft/demo/domain/MNISTDataRepository.kt]: Yeah will do

**Pull request merged**